src/transformers/data/data_collator.py

```diff
@@ -34,30 +34,50 @@ def default_data_collator(features: List[InputDataClass]) -> Dict[str, torch.Ten
     # So we will look at the first element as a proxy for what attributes exist
     # on the whole batch.
     first = features[0]
+    batch = {}
+
+    # Note: below there are a few mysterious torch.stack([torch.tensor(f) for f ...]).
+    # This is because torch.tensor does not accept a list of tensors, to deal with the case the f already are tensors
+    # and not lists of ints.
+
+    if isinstance(first, dict):
```

Comment:
This code looks very duplicated but I don't know if we have an easy + performant way of removing the duplication.
Maybe just doing:
```
d = [vars(f) for f in features]
```
at the top and only handle dicts?

Reply:
Oh smart, should indeed work to remove the dupe code.

Reply:
I just hope the perf penalty is acceptable, given that this is on the "hot path" of each batch during training. cc @mfuntowicz

Reply:
Why would it be longer that the current loop over keys of `var(first)` then `getattr`?